bluemira/base/parameter_frame/_frame.py

```diff
@@ -115,14 +123,133 @@ def to_dict(self) -> Dict[str, Dict[str, Any]]:
+
+    @staticmethod
+    def _fix_combined_units(unit: pint.Unit) -> pint.Unit:
+        dim_keys = list(combined_unit_dimensions.keys())
+        dim_val = list(combined_unit_dimensions.values())
+        with suppress(ValueError):
+            return pint.Unit(
+                combined_unit_defaults[dim_keys[dim_val.index(unit.dimensionality)]]
+            )
+        return pint.Unit(unit)
+
+    @classmethod
+    def _fix_weird_units(
+        cls, modified_unit: pint.Unit, orig_unit: pint.Unit
+    ) -> pint.Unit:
+        """
+        Essentially a crude unit parser for when we have no dimensions
+        or non-comutative dimensions
+        """
+        unit_str = f"{orig_unit:D}"
+
+        ang_unit = [ang for ang in ANGLE_units if ang in unit_str]
+        if len(ang_unit) > 1:
+            raise ValueError(f"No...{orig_unit}")
+        elif len(ang_unit) == 1:
+            ang_unit = ang_unit[0]
+        else:
+            ang_unit = None
+
+        fpy = "full_power_year" in unit_str
+        dpa = "displacements_per_atom" in unit_str
+
+        if not (fpy or dpa) and ang_unit is None:
+            return pint.Unit(modified_unit)
+
+        if modified_unit == pint.Unit("dimensionless") and dpa:
+            new_unit = pint.Unit(
+                f"displacements_per_atom^{-1 if len(unit_str.split('/')) > 1 else 1}"
+            )
+        elif fpy:
+            if not modified_unit.dimensionality.keys() - ["[time]"]:
+                expon = list(modified_unit.dimensionality.values())[0]
+                if dpa:
+                    dpa_expon = -1 if len(unit_str.split("/")) > 1 and expon == 1 else 1
+                    new_unit = pint.Unit(f"dpa^{dpa_expon}.fpy^{expon}")
+                else:
+                    new_unit = pint.Unit(f"fpy^{expon}")
+            else:
+                unit_list = unit_str.split("/")
+                if dpa:
+                    if "displacements_per_atom" in unit_list[0]:
+                        dpa_str = "dpa."
+                    else:
+                        dpa_str = "dpa^-1."
+                else:
+                    dpa_str = ""
+
+                dim = dict(modified_unit.dimensionality)
+                if "full_power_year" in unit_list[0]:
+                    dim["[time]"] -= 1
+                    new_unit = pint.Unit(
+                        f"{dpa_str}fpy.{cls._fix_combined_units(cls._remake_units(dim))}"
+                    )
+                else:
+                    dim["[time]"] += 1
+                    new_unit = pint.Unit(
+                        f"{dpa_str}{cls._fix_combined_units(cls._remake_units(dim))}/fpy"
+                    )
+        else:
+            new_unit = modified_unit
+
+        # Deal with angles
+        if ang_unit:
+            return cls._fix_angle_units(new_unit, orig_unit, ang_unit)
+        else:
+            return new_unit
+
+    @staticmethod
+    def _fix_angle_units(modified_unit, orig_unit, angle_unit):
```

Comment:
This could probably do with some documentation on what we mean by 'fix' and why we need to 'fix' angles.